src/Downloader.ts

```diff
@@ -323,10 +329,14 @@ class Downloader {
 
     return articleRenderer.render({
       data,
+      redisStore,
+      webp,
```

Comment:
I'm a bit concerned about the multiple responsibilities of `saveArticles()`. For instance, `const { jsModuleDependencies, cssModuleDependencies } = await saveArticles(zimCreator, downloader, redisStore, dump)` [here](https://github.com/openzim/mwoffliner/blob/main/src/mwoffliner.lib.ts#L422C5-L422C121) allows to get js and CSS dependencies and at the same time write HTML into zim and modify dump instance while render process. Maybe this is also a symptom of weak architecture here. As for DumpOpts, I agree that this is a better idea to pass options instead of a full dump instance

Reply:
I agree, I would really like to have this js/css module handling done in a dedicated method

Reply:
OK, to me it sounds we should almost have a #3 of rearchitecture to get rid of `saveArticles()` and put the main routine at a better place. That said, unsure about how the solution should look like and when we should do this. At this stage, I propose to just keep things like they are as long as a solution does not appear clearly. Once this PR merge, we should really focus then to fix the bugs and get again the mobile-html stuff working finel. That's the priority. We can close this discussion.

Reply:
Agree, lets focus on bugs and mobile-html first.